STYLEGUIDE.md

```diff
@@ -0,0 +1,215 @@
+
+All conda commands will be typed in the Terminal window.
+
+```commandline
+conda create --name pcf-local python==3.6
+```
+
+*PCF requires Python version 3.6+*
+
+### Activate virutal env
+```commandline
+source activate pcf-local
+```
+
+### Deactivate virtual env
+```commandline
+source deactivate
+```
+
+### View list of exisiting virtual envs
+```commandline
+conda info --envs
+```
+
+## <a name="imports">Imports</a>
+------
+Refrain importing entire modules when only 1 object from the module is needed.
+
+**Don't:**
+```python
+import boto3, json, logging
+import pcf.core.PCF_exceptions 
+```
+
+**Do:**
+```python
+import boto3
+import json
+import logging
+
+from pcf.core.pcf_exceptions import NoResourceException
+```
+
+## <a name="naming">Naming</a>
+------
+PCF typically follows the same naming conventions as <a href="https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html">PEP8 standards</a>.
+
+### <a name="variables">Variables</a>  
+```python
+# Lowercase multi-word separated by an underscore
+my_greeting = "hello"
+```
+
+```python
+# Non-public instance variables should begin with a single underscore
+_client = ""
+```
+
+### <a name="classes">Classes</a>
+```python
+# UpperCaseCamelCase convention
+class LambaFunction():
+    # class
+
+class AWSResource():
+    # class
+
+class EC2Instance():
+    # class
+```
+
+### <a name="funcs">Fucntions/Methods</a>
+```python
+# Lowercase multi-word separated by an underscore
+def get_status(self):
+    pass
+```
+```python
+# Non-public methods should begin with a single underscore
+def _terminate(self):
+    pass
+```
+
+## <a name="spacing">Spacing</a>
+------
+* 4 spaces = 1 indent
+* Leave a single blank line after function/method declarations to aid in visual clarity and organization
+
+## <a name="strings">Strings</a>
+------
+### <a name="quotes">Quotes</a>  
+
+    In Python single quotes and double quotes are used interchangablely. We will stick to double quotes for consistency.  
+
+**Don't:**  
+```python
+nameOfSchool = 'Marshall'
+nameOfOffice = "Clarendon"
+```  
+
+**Do:**  
+```python
+nameOfSchool = "Marshall"
+nameOfOffice = "Clarendon"
+```   
+
+### <a name="docstr">Docstrings</a>  
+Docstrings act as documentation for method definitions of a class within PCF. Always use """triple double quotes""" around docstrings. For multi-line docstrings, place the closing qoutes on a line by itself.
+
+The docstrings should give a brief description of what the method/class is doing, explain arguments and their type, if any, and list what the method/class returns.
+
+For more info see the <a href="https://www.python.org/dev/peps/pep-0257/#what-is-a-docstring">docs</a>.
+
+**Example:**
+```python
+def is_state_equivalent(self, state1, state2):
+    """
+    Determines if states are equivalent. Uses equivalent_states defined in the Glacier class.
+
+    Args:
+        state1 (State):
+        state1 (State):
+
+    Returns:
+        bool
+    """
+```
+
```

Comment:
Let's add something about how we won't merge something in if the classes and methods are not documented this way.